Ginger/Ginger/Run/RunSetsExecutionsHistoryPage.xaml.cs

```diff
@@ -333,40 +416,296 @@ private async void LoadExecutionsHistoryData()
-            if (RunsetConfig != null)
+            ObservableList<RunSetReport> executionsHistoryListSortedByDate = new ObservableList<RunSetReport>();
+            if (mExecutionsHistoryList != null && mExecutionsHistoryList.Count > 0)
             {
-                runsetsReport = new GingerRemoteExecutionUtils().GetRunsetExecutionInfo(WorkSpace.Instance.Solution.Guid, RunsetConfig.Guid);
+                IEnumerable<RunSetReport> sortedAndFilteredExecutionHistoryList = mExecutionsHistoryList
+                   .Where(report => report.RunSetExecutionStatus != eRunStatus.Automated)
+                   .OrderByDescending(item => item.StartTimeStamp);
+
+                foreach (RunSetReport runSetReport in sortedAndFilteredExecutionHistoryList)
+                {
+                    runSetReport.StartTimeStamp = runSetReport.StartTimeStamp.ToLocalTime();
+                    runSetReport.EndTimeStamp = runSetReport.EndTimeStamp.ToLocalTime();
+                    executionsHistoryListSortedByDate.Add(runSetReport);
+                }
             }
-            else
+
+            xGridExecutionsHistory.DataSourceList = executionsHistoryListSortedByDate;
+            xGridExecutionsHistory.Visibility = Visibility.Visible;
+            Loading.Visibility = Visibility.Collapsed;
+        }
+        /// <summary>
+        /// Loads remote data from GraphQL and updates the execution history page.
+        /// </summary>
+        private async Task LoadRemoteDataFromGraphQL()
+        {
+            try
+            {
+                int recordLimit = (int)xPageSizeComboBox.SelectedItem;
+                graphQlClient.ResetPagination();
+
+                response = RunsetConfig == null
+                    ? executionReportGraphQLClient.ExecuteReportQuery(recordLimit, WorkSpace.Instance.Solution.Guid)
+                    : executionReportGraphQLClient.ExecuteReportQuery(recordLimit, WorkSpace.Instance.Solution.Guid, RunsetConfig.Guid);
+
+                var data = await response;
+                AddRemoteDataToList(data);
+                UpdatePageInfo("firstPage");
+                UpdateButtonStates();
+            }
+            catch (Exception ex)
+            {
+                Reporter.ToLog(eLogLevel.ERROR, $"Error occurred while connecting remote.", ex);
+                Reporter.ToUser(eUserMsgKey.RemoteExecutionHistoryEndPoint);
+                
+            }
+        }
+
+        /// <summary>
+        /// Handles pagination button click events.
+        /// </summary>
+        private async Task AddDataOnButtonClick(int recordLimit, string pageButton = null, string endCursor = null, string startCursor = null, bool firstPage = false, bool lastPage = false, bool afterOrBefore = true)
+        {
+            try
             {
-                runsetsReport = new GingerRemoteExecutionUtils().GetSolutionRunsetsExecutionInfo(WorkSpace.Instance.Solution.Guid);
+                response = RunsetConfig == null
```

Comment:
Try to extract function for this code as same code is being executed in LoadRemoteDataFromGraphQL()

Reply:
done